common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -913,6 +998,50 @@ public UTF8String subStringIndex(UTF8String delim, int count) {
     }
   }
 
+  public UTF8String collationAwareSubStringIndex(UTF8String delim, int count, int collationId) {
```

Comment:
I think we now have a lot of new methods 
consider adding comments such as: https://github.com/apache/spark/pull/45791/files#r1549415554